### PR TITLE
[csl] avoid OT version dependencies in platform config files

### PR DESCRIPTION
**Pull request by canisLupus1313**

### CODE REVIEW

File: src/core/mac/sub_mac.cpp [hunk 1]
```diff
@@ -105,15 +105,15 @@ otRadioCaps SubMac::GetCaps(void) const
     caps |= OT_RADIO_CAPS_ENERGY_SCAN;
 #endif
 
-#if OPENTHREAD_CONFIG_MAC_SOFTWARE_TX_SECURITY_ENABLE
+#if OPENTHREAD_CONFIG_MAC_SOFTWARE_TX_SECURITY_ENABLE && (OPENTHREAD_CONFIG_THREAD_VERSION >= OT_THREAD_VERSION_1_2)
```
Comment (xiaonanln) [inline src/core/mac/sub_mac.cpp:108]: Should also change the default value to 1 in `openthread-core-simulation-config.h` ?

Comment (simonlingoogle) [inline src/core/mac/sub_mac.cpp:108]: Should also change the default value to 1 in `examples/platforms/simulation/openthread-core-simulation-config.h` ?

https://github.com/openthread/openthread/blob/4484042ced953b9d24a47cc8f3a04ea0e531e18c/examples/platforms/simulation/openthread-core-simulation-config.h#L118

**Review by simonlingoogle — COMMENTED** [review]
S

File: src/core/mac/sub_mac.cpp (hunk 1 above)
Comment (canisLupus1313) [reply src/core/mac/sub_mac.cpp:108]: I think You are wright. Fixed this.

**Review by edmont — APPROVED** [review]

**Review by simonlingoogle — APPROVED** [review]
LGTM 👍

**Review by Irving-cl — APPROVED** [review]
LGTM

**Review by jwhui — APPROVED** [review]
Thanks! 👍🏼

**Pull request merged**